malicious-site-protection/malicious-site-protection-impl/src/main/kotlin/com/duckduckgo/malicioussiteprotection/impl/domain/RealMaliciousSiteProtection.kt

```diff
@@ -41,57 +46,60 @@ class RealMaliciousSiteProtection @Inject constructor(
     private val maliciousSiteProtectionRCFeature: MaliciousSiteProtectionRCFeature,
 ) : MaliciousSiteProtection {
 
-    override suspend fun isMalicious(url: Uri, confirmationCallback: (isMalicious: Boolean) -> Unit): IsMaliciousResult {
+    override suspend fun isMalicious(
+        url: Uri,
+        confirmationCallback: (confirmedResult: MaliciousStatus) -> Unit,
+    ): IsMaliciousResult {
         Timber.tag("MaliciousSiteProtection").d("isMalicious $url")
 
         if (!maliciousSiteProtectionRCFeature.isFeatureEnabled()) {
             Timber.d("\uD83D\uDFE2 Cris: should not block (feature disabled) $url")
-            return IsMaliciousResult.SAFE
+            return ConfirmedResult(Safe)
         }
 
-        val hostname = url.host ?: return IsMaliciousResult.SAFE
+        val hostname = url.host ?: return ConfirmedResult(Safe)
         val hash = messageDigest
             .digest(hostname.toByteArray(Charsets.UTF_8))
             .joinToString("") { "%02x".format(it) }
         val hashPrefix = hash.substring(0, 8)
 
         if (!maliciousSiteRepository.containsHashPrefix(hashPrefix)) {
             Timber.d("\uD83D\uDFE2 Cris: should not block (no hash) $hashPrefix,  $url")
-            return IsMaliciousResult.SAFE
+            return ConfirmedResult(Safe)
         }
-        maliciousSiteRepository.getFilters(hash)?.let {
-            for (filter in it) {
-                if (Pattern.compile(filter.regex).matcher(url.toString()).find()) {
-                    Timber.d("\uD83D\uDFE2 Cris: shouldBlock $url")
-                    return IsMaliciousResult.MALICIOUS
-                }
+        maliciousSiteRepository.getFilters(hash)?.forEach { filterSet ->
+            filterSet.filters.firstOrNull {
+                Pattern.compile(it.regex).matcher(url.toString()).find()
+            }?.let {
+                Timber.d("\uD83D\uDFE2 Cris: shouldBlock $url")
```

Comment:
NIT: do we have a Tag/Prefix you want to use from now on for logs? Probably better to start using it now, in case we forget to clean the logs with Cris. 

Maybe just use `MaliciousSiteDetector` or MaliciousSite`?

Reply:
Yes, we can do that. My plan was to leave them for testing until the very end, but since we started merging a few things to develop already, I can update the tag